<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>bogui</class>
 <widget class="QMainWindow" name="bogui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>740</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>bogui</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPlainTextEdit" name="edtStack">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>290</y>
      <width>581</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="edtCode">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>20</y>
      <width>241</width>
      <height>251</height>
     </rect>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="edtScreen">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>340</y>
      <width>771</width>
      <height>231</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="controls" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>580</y>
      <width>471</width>
      <height>121</height>
     </rect>
    </property>
    <widget class="QPushButton" name="btnStep">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>0</y>
       <width>80</width>
       <height>61</height>
      </rect>
     </property>
     <property name="text">
      <string>Step</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnStart">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>0</y>
       <width>80</width>
       <height>61</height>
      </rect>
     </property>
     <property name="text">
      <string>Start</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnCompile">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>80</width>
       <height>61</height>
      </rect>
     </property>
     <property name="text">
      <string>Compile</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnStop">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>60</y>
       <width>80</width>
       <height>61</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnRun">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>80</width>
       <height>61</height>
      </rect>
     </property>
     <property name="text">
      <string>Compile
and Run</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="grpSecurity">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>0</y>
       <width>211</width>
       <height>111</height>
      </rect>
     </property>
     <property name="title">
      <string>Counter Measures:</string>
     </property>
     <widget class="QComboBox" name="canaries">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>40</y>
        <width>121</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frame">
       <bool>true</bool>
      </property>
      <item>
       <property name="text">
        <string>none</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>terminator</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>random</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>random xor</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="lblcanaries">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>62</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Canaries:</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="rocode">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>191</width>
        <height>31</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Read-Only Code Segment</string>
      </property>
      <property name="tristate">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="bounds">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>151</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Bounds Checking</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblchange">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>201</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>(will change at next compile)</string>
      </property>
      <property name="scaledContents">
       <bool>false</bool>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="indent">
       <number>-1</number>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QTableWidget" name="tblCodeSeg">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>300</y>
      <width>241</width>
      <height>271</height>
     </rect>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>80</number>
    </attribute>
    <attribute name="verticalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <column>
     <property name="text">
      <string>addr</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>instruction</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>opr1</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>opr2</string>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="lblCodeSegment">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>280</y>
      <width>181</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Code Segment of Memory:</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblCodeInput">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>111</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Write code here:</string>
    </property>
   </widget>
   <widget class="QWidget" name="inputs" native="true">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>580</y>
      <width>541</width>
      <height>121</height>
     </rect>
    </property>
    <widget class="QPlainTextEdit" name="edtInput">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>541</width>
       <height>61</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnAppendInt">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>100</y>
       <width>80</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>full int</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinChar">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>100</y>
       <width>60</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>65000</number>
     </property>
    </widget>
    <widget class="QPushButton" name="btnAppendChar">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>100</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>single character</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblappends">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>100</y>
       <width>421</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Append a character with value:                   as a                      or a</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
    </widget>
    <widget class="QLabel" name="lblinput">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>541</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Set the contents of the input buffer here. Read operations take the first line,
     readn operations take the first n characters.</string>
     </property>
    </widget>
    <widget class="QComboBox" name="inputsamples">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>20</y>
       <width>231</width>
       <height>21</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Input Samples</string>
      </property>
     </item>
    </widget>
    <zorder>lblappends</zorder>
    <zorder>edtInput</zorder>
    <zorder>btnAppendChar</zorder>
    <zorder>lblinput</zorder>
    <zorder>inputsamples</zorder>
    <zorder>spinChar</zorder>
    <zorder>btnAppendInt</zorder>
   </widget>
   <widget class="QLabel" name="lblscreen">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>320</y>
      <width>311</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Compiler and Runtime Output is shown here:</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tblStack">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>20</y>
      <width>771</width>
      <height>261</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string>frame</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>var name</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>var sizes</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>var data</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>canary</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>prev base</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>prev top</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>ret addr</string>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="lblStack">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>0</y>
      <width>311</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>The Contents of the stack are shown below:</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblNext">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>290</y>
      <width>201</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Next Instruction to execute:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="codesamples">
    <property name="geometry">
     <rect>
      <x>114</x>
      <y>0</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Code samples...</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
